tools/espota.py

```diff
@@ -94,8 +94,8 @@ def serve(remoteAddr, localAddr, remotePort, localPort, password, filename, comm
   file_md5 = hashlib.md5(f.read()).hexdigest()
   f.close()
   logging.info('Upload size: %d', content_size)
-  message = '%d %d %d %s\n' % (command, localPort, content_size, file_md5)
 
+  message = '%d %d %d %s\n%s\n' % (command, localPort, content_size, file_md5, localAddr)
```

Comment:
I am worried that this will break existing deployments since the wire protocol is now different.  The existing code will not read the extra IP address sent afterwards, but instead interpret it as part of the actual update and fail.

Can someone confirm that a build of an app w/the main branch code works fine with this new updater (i.e. an update of a deployed system).?

Reply:
It’s backwards compatible because the (old and new) client luckily only reads the md5 hash until encountering `\n`.

So the newly added ip adress would sit in the udp message buffer but that‘s it. 

The update itself happens via a secondary tcp connection and is not affected.

Reply:
Thanks for this clarification :+1: